### PR TITLE
[WIP:] Slackdemo2

### PR DESCRIPTION
**Pull request by JohnONolan**
@ErisDS 
- Cleaned up icon
- Added short link to my.slack.com for webhook creation
- Removed channel config - this is confusing, seeing as you set it up on slack.com already
- Removed username config - there's no reason this shouldn't just be "Ghost" at the moment
- Cleaned up some markup - there is unfortunately a lot more to do here.
- Replaced pixelated icon with new hires icon

Still needs doing:
- There should be no "activate" button on the first screen. It should just be a right > chiclet. Same as post-settings menu meta data. This does not need to change state.
- I recommend stripping the concept of "activate" completely 

Have to get on a plane now so I can't finish this list

### CODE REVIEW

**Comment by ErisDS** [comment]
> There should be no "activate" button on the first screen. It should just be a right > chiclet. Same as post-settings menu meta data. This does not need to change state.

I think the whole bar is meant to be a link as well, rather than just the arrow? That's how the PSM functions so I think it should be the same.

> I recommend stripping the concept of "activate" completely

It seems to me that the url should be empty to start, rather than `/` and that we should therefore base active/inactive on whether a URL is set.

I think for this version, with only one field, it does make sense to drop the activate/deactivate button. Ideally we would have a way other than the URL to indicate whether or not the feature was active so that we can do things like - deactivate if the ping fails 5 times (URL is probably wrong) and to have a consistent way across integrations / apps to determine if the functionality is supposed to be "on" else the software will be doing more work (using more resources) than needed. Still this will probably all fall out much more naturally when we have more than one of these in place and we can figure out a good framework for them.

**Comment by ErisDS** [comment]
FYI I'm working on this and I think I have it working ok, but probably not the best code 🙃

I think I can push direct to this branch

**Comment by ErisDS** [comment]
I've pushed a commit that removes the whole concept of activate/deactivate everywhere.

`isActive` is now a computed property based on whether the URL is empty or not. The default url is now empty instead of `/`. 

I also set the url field to have a placeholder of `https://hooks.slack.com/services/...` as there is now default value.

I also did a bit of cleanup around the username, it will now always be 'Ghost' just like the icon will always be the ghost emoji 👻 . There is a way to set this when you configure your webhook, just as there is with the channel, however our settings will always override this.

I'm not sure if this is really the right thing? We could also not include them at all, and the messages would use whatever the user configures in slack. Either way works, one way we get consistency, the other way the user gets control and can configure different things e.g. if they have more than one blog posting into the same slack channel (edge case, but we have that 😉 ).

**Comment by JohnONolan** [comment]
Cleaned some more styles - this is looking way better now. I think it needs a save button on the detail state? I know it doesn't *need* one but it's unclear if state has been saved or not. 

Icon in navbar on the left needs to change - that's a me thing - will come back to it!

Everything else is looking good I think. @kevinansfield if you can spot any UX improvements here - would definitely be welcome

**Comment by ErisDS** [comment]
As we're moving away from save buttons, it seems weird to me to add one for this. I know what you mean about it being unclear - it's the same on the tags page, labs page, etc In this case, I think it might work to have some sort of "going green" validation? Either an icon like on signup/setup etc or a green border or checkmark next to the URL field to demonstrate that it has been validated / "accepted".

There is an error message that appears if it isn't valid, so this would fit with what we already have? Just a suggestion as that was what came to my mind before you said about a save button.

One thing I noticed as well was that if you hit enter in the url input, the form submits and this causes a test notification, rather than a save, which was what I would have expected.

**Comment by JohnONolan** [comment]
@ErisDS icon updated - I think I'm done. I agree with everything you said about save button, but the fact remains that we don't *have* anything to replace it sanely. I don't think there's a better option at the moment than just having a save button, until we actually have something better.

@kevinansfield Not related to this PR I don't think - but we seem to have lost a lot of styling on the search box. I've included a few fixes in the latest commit, but there's still more that doesn't look right. Eg there's no border colour on:focus, and the "no results" message formatting has broken.

*Before:*
image

*After:*
image

**Comment by ErisDS** [comment]
@kevinansfield can you look at adding the save button? I think it is beyond my ember fu.

@johnonolan can you go to settings on your repo and add Kevin as a collaborator? Then we don't need a new PR again 😉

**Comment by JohnONolan** [comment]
Done!

**Comment by kevinansfield** [comment]
Just to be clear, once the save button is in are we getting rid of the save on focus-out or keeping that? (focus-out will cause the save button to spin so there is some indication that saving is happening)

Re the search input styling I'll tackle that in a separate PR.

**Comment by JohnONolan** [comment]
@kevinansfield either is fine w/ me

File: core/client/app/styles/components/power-select.css [hunk 1]
```diff
@@ -106,3 +110,14 @@
     background: color(var(--blue) alpha(-85%));
     color: var(--darkgrey);
 }
+
+/*
+    HACK: ember-power-select has no separate class for the loading message
+    Issue: https://github.com/cibernox/ember-power-select/issues/479
+ */
+.ember-power-select-dropdown > .ember-power-select-options > .ember-power-select-option:first-of-type,
```
Comment (kevinansfield) [inline core/client/app/styles/components/power-select.css]: The PR I submitted to ember-power-select has been merged and should be in a new release due this weekend so hopefully this particular piece of ugliness won't be in the codebase for too long.

**Comment by ErisDS** [comment]
Closing in favour of #6795

**Pull request closed without merging**